File: June-18/01-06-18/python/01-Compare-String.py
N = 10**6 + 2
dp = [0]*N
cnt = 0




n, q = map(int,input().split())
str1 = list(input())
str2 = list(input())
for i in range(0, n):
    if(str1[i] != str2[i]):
        dp[i] = 1
        cnt += 1

flag = False
for i in range(0,q):
    x = int(input())
    x -= 1
    
    if(dp[x]):
        cnt -= 1
    else:
        if(str1[i] == '0'):
            flag = True
    
    if(cnt == 0 and flag == False):
        print("YES")
    else:
        print("NO")

File: June-18/01-06-18/c++/01-Compare-String.cpp
#include<bits/stdc++.h>

using namespace std;
map<int,int> dp;
int cnt = 0;

int cmp(string str1, string str2)
{
    int n1 = str1.length();
    
    for(int i = 0; i < n1; i++)
    {
        if(str1[i] != str2[i])
        {
            dp[i] = 1;
            cnt += 1;   
        }
    }
    return 1;
}

int main()
{
    int n, q;
    cin >> n >> q;
    string str1, str2;
    cin >> str1 >> str2;
    int  x = cmp(str1, str2);
    bool flag = false;
    while(q--)
    {
        scanf("%d", &x);
        x--;
        if(dp[x])
            cnt -= 1;
        else
        {
            if(str1[x] == '0')
                flag = true;
        }
        
        if(cnt == 0 && !flag)
            printf("YES\n");
        else
            printf("NO\n");
    }
}
